src/pass_manager.cpp

```diff
@@ -72,20 +71,19 @@ struct module_pm : module_pass_manager
     virtual void run_pass(const pass& p) override
     {
         assert(mod);
-        trace("Module: ", mod->name(), ", Pass: ", p.name());
         assert(mod->validate() == mod->end());
         p.apply(*this);
-        trace(*mod);
-        validate_pass(*mod, p, *t);
+        trace(p.name(), *mod);
+        validate_pass(*mod, p);
     }
 };
 
 module& get_module(module_pass_manager& mpm) { return mpm.get_module(); }
 
 void run_passes(module& mod, const std::vector<pass>& passes, tracer trace)
 {
-    if(enabled(MIGRAPHX_TRACE_PASSES{}))
-        trace = tracer{std::cout};
+    if(enabled(MIGRAPHX_TRACE_PASSES{}) and not trace.enabled())
+        trace = tracer{mod.name() + "_passes"};
```

Comment:
So this looks like `MIGRAPHX_TRACE_PASSES` will dump to a file instead of stdout.

Reply:
Yes, by design.